src/blog/2023/04/securing-node-red-in-production.md

```diff
@@ -0,0 +1,150 @@
+
+You might also want to consider turning off the editor interface once you are happy with your flows. This can make it a little harder to make changes to your project but it also gives you peace of mind that nobody has accidentally or deliberately changed your flows. You can turn off the editor interface as follows.
+
+Edit your settings.js file as explained above, look for the following section:
+
+![The setting file before turning off the editor](./images/editor-on.png "The setting file before turning off the editor")
+
+All you need to do is uncomment the bottom line then change the value from false to true, once done it should look something like this:
+
+![The setting file after turning off the editor](./images/editor-off.png "The setting file after turning off the editor")
+
+Now restart Node-RED as covered above, then try accessing your Node-RED instance again. You will no longer be able to edit or view your flows.
+
+Using these two features, we now have much better control over who can access the design interface for Node-RED.
+
+## Traffic to your Node-RED instance is unencrypted
+
+Hopefully, we all know the importance of encrypting your connections between devices to stop people intercepting your traffic. This isn't a huge concern when working on your home LAN but what if you want to access your Node-RED instance from a remote location? 
+
+There are two obvious options, HTTPS, and a VPN (Virtual Private Network).
+
+We could setup your Node-RED traffic to run over HTTPS, this solution ensures that all traffic to and from your Node-RED is encrypted. The downside to this approach is it's quite complex to set up. We will need to have a domain name, open up ports on our LAN's firewall, use a HTTPS certificate provider and then make sure we remember to renew the certificates as needed. It's doable if you are comfortable with those concepts (I covered how to do this as part of my blog [hosting FlowForge on Google Cloud](/blog/2022/12/flowforge-gcp-https-set-up/)) but there is an easier way to get started, using a VPN.
+
+A VPN provides a lot of security advantages depending on which you are using and how it is configured. To secure my traffic I'm going to use a great service call [Tailscale](https://tailscale.com/) which is free for personal projects.
+
+I'm going to install Tailscale on the Raspberry Pi I'm running Node-RED on as well as any other devices I want to access my project from. Once that's done I can access Node-RED from anywhere with internet access but more importantly the traffic to and from my devices is also encrypted.
+
+Before we start, it's important to remember that a VPN is only as secure as the company who runs it. You should always consider if you trust the VPN provider as they could potentially access your devices. I trust Tailscale but please do your own research before using a VPN provider.
```

Comment:
Is this really needed?

Reply:
It's not clear what 'this' is referring to, can you be specific please?

Reply:
The paragraph about trusting your vpn provider.

Reply:
I think so yes, I think we should recommend a VPN without stating the risks of using them (or any VPN provider).

Reply:
OK, it just sounds a bit patronizing to me but your call.